### PR TITLE
[FEATURE] Implemented new status for pending cart

### PR DESCRIPTION
**Pull request by turbo-ele**

### CODE REVIEW

File: Api/Order.php [hunk 1]
```diff
@@ -1134,4 +1142,46 @@ public function setHasChangedPrices($hasChangedPrices)
 
         return $this;
     }
+
+    /**
+     * Set status code of cart
```
Comment (notsure) [inline Api/Order.php]: get status code

File: Api/Order.php [hunk 2]
```diff
@@ -1134,4 +1142,46 @@ public function setHasChangedPrices($hasChangedPrices)
...
+     */
+    public function addCartErrorCode($statusCode)
+    {
+        $this->cartStatusCodes[] = $statusCode;
+
+        return $this;
+    }
+
+    /**
+     * If cart is in cart_pending state
```
Comment (notsure) [inline Api/Order.php]: Returns if cart is in ...

File: Cart/CartManager.php [hunk 3]
```diff
@@ -321,16 +343,34 @@ public function submit($user, $locale, &$orderWasSubmitted = true, &$originalCar
...
+     *
+     * @param UserInterface $user
+     * @param ApiOrderInterface $cart
+     * @param string $locale
+     *
+     * @throws OrderException
+     */
+    protected function checkIfCartIsValid(UserInterface $user, ApiOrderInterface $cart, $locale)
+    {
+        if (count($cart->getItems()) < 1) {
```
Comment (notsure) [inline Cart/CartManager.php:356]: no else?

Comment (turbo-ele) [inline Cart/CartManager.php:356]: no, it's just an error if no items are available

**Comment by notsure** [comment]
ack

**Pull request merged**